server/index.js

```diff
@@ -23,103 +23,128 @@ const open = require('open');
 const portfinder = require('portfinder');
 const crypto = require("crypto");
 const fse = require('fs-extra');
+const { createHttpTerminator } = require('http-terminator');
 
+const DEFAULT_PORT = '9000';
 
-async function start(portPreferred) {
-    let port = process.env.ASSET_COMPUTE_DEV_PORT || '9000';
-    if (!isNaN(portPreferred)) {
-        port = portPreferred;
+class DevtoolServer {
+
+    async run (portPreferred) {
```

Comment:
Nitpick: preferredPort sounds more natural?

Reply:
Also, I think `run` needs some comment-documentation to clarify the implications of different types for the port. According to the code, `this.port`can be a string or a number. And it seems there is a distinction between `port`and `pipe` also in error messages.

Reply:
Yes, the port can be either string or number. I added test cases for each of the different ways the port can be defined